exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -1543,7 +1545,14 @@ func (ku *Kucoin) GetFuturesContractDetails(ctx context.Context, item asset.Item
 		if contracts[i].IsInverse {
 			contractSettlementType = futures.Inverse
 		}
-		timeOfCurrentFundingRate := time.Now().Add((time.Duration(contracts[i].NextFundingRateTime) * time.Millisecond) - ku.Features.Supports.FuturesCapabilities.FundingRateFrequency).Truncate(time.Hour).UTC()
+		var fri time.Duration
+		if len(ku.Features.Supports.FuturesCapabilities.SupportedFundingRateFrequencies) == 1 {
+			// can infer funding rate interval from the only funding rate frequency defined
+			for k := range ku.Features.Supports.FuturesCapabilities.SupportedFundingRateFrequencies {
+				fri = k.Duration()
```

Comment:
Suggestion: Can use the key to access instead of ranging to make sure there are no future change issues. This will randomise access if more than 1 are added. Also the check above will just set this as zero if another one is aded. 
```go
k, ok := ku.Features.Supports.FuturesCapabilities.SupportedFundingRateFrequencies[kline.Eighthour]
```

Reply:
I understand your point, but this code is specifically added because there is only one supported like the comment and the if statement says. If it is beyond that, more work needs to be done, either by retrieving data like for binance or otherwise. Do you want me to error if the length is > 1?

Reply:
Nar just leave it, I get what you are doing now.